### PR TITLE
Move and rename validation step's types

### PR DESCRIPTION
**Pull request by loic-sharma**
# Background

This change moves and then renames the following types from ServerCommon to NuGet.Jobs:

Old name | New name
-- | --
`IProcessor` | `INuGetProcessor`
`IValidationRequest` | `INuGetValidationRequest`
`IValidationResult` | `INuGetValidationResponse`
`IValidator` | `INuGetValidator`
`ValidationResult` | `NuGetValidationResponse`

⚠**NOTE**: I moved these types to the `Validation.Common.Job` project instead of the `NuGet.Services.Validation.Orchestrator` project. This is intentional as these types were referenced in `Validation.Common.Job` (for example, see [this](https://github.com/NuGet/NuGet.Jobs/blob/be8293c63832830dec04ab28e1d9af89e6a3df1f/src/Validation.Common.Job/Storage/IValidatorStateService.cs#L20)).

This also renames the Orchestrator's `ValidationRequest` type to `NuGetValidationRequest`.

Part of: https://github.com/NuGet/Engineering/issues/3614

# How to review

1. Review the corresponding ServerCommon change: https://github.com/NuGet/ServerCommon/pull/356.
2. You can skip this change's first commit as it moves the types without any changes: https://github.com/NuGet/NuGet.Jobs/commit/b30cc28b7833fa0b215816138a6fe6348e2184cc
3. Please thoroughly review this change's second commit as it renames the types: https://github.com/NuGet/NuGet.Jobs/pull/958/commits/b934e0ee46cb59656c62a13c860801ed87021005
4. You can skim this change's third commit as it updates the rest of the repository to reflect the new types: https://github.com/NuGet/NuGet.Jobs/pull/958/commits/b99261efd034910e920d9a3e7858fbe1a4c82a13

# Verification

The NuGet package and symbol Orchestrator have been deployed to DEV USNC and DEV USSC. I verified all end-to-end tests passed:

* Package orchestrator release: https://devdiv.visualstudio.com/DevDiv/_releaseProgress?_a=release-pipeline-progress&releaseId=941153
* Symbol orchestrator release: https://devdiv.visualstudio.com/DevDiv/_releaseProgress?_a=release-pipeline-progress&releaseId=941158

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <!-- NuGet dependencies shared across projects -->
   <PropertyGroup>
-    <ServerCommonPackageVersion>2.80.0</ServerCommonPackageVersion>
+    <ServerCommonPackageVersion>2.82.0-loshar-uvp-rename-4411505</ServerCommonPackageVersion>
```
Comment (loic-sharma) [inline Directory.Build.props:5]: TODO: https://github.com/NuGet/ServerCommon/pull/356

File: src/NuGet.Jobs.GitHubIndexer/NuGet.Jobs.GitHubIndexer.csproj [hunk 2]
```diff
@@ -93,7 +93,7 @@
       <Version>0.32.0</Version>
     </PackageReference>
     <PackageReference Include="NuGetGallery.Core">
-      <Version>4.4.5-dev-4220086</Version>
+      <Version>4.4.5-loshar-update-servercommon-4411704</Version>
```
Comment (loic-sharma) [inline src/NuGet.Jobs.GitHubIndexer/NuGet.Jobs.GitHubIndexer.csproj:96]: TODO: https://github.com/NuGet/NuGetGallery/pull/8403

**Review by zhhyu — APPROVED** [review]

**Review by skofman1 — APPROVED** [review]

**Review by agr — APPROVED** [review]

File: tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationProviderFacts.cs [hunk 3]
```diff
@@ -157,39 +157,39 @@ public BaseFacts()
...
+        public class TestNuGetValidator : INuGetValidator
         {
-            public Task CleanUpAsync(IValidationRequest request)
+            public Task CleanUpAsync(INuGetValidationRequest request)
             {
                 throw new NotImplementedException();
             }
 
-            public Task<IValidationResult> GetResultAsync(IValidationRequest request)
+            public Task<INuGetValidationResponse> GetResultAsync(INuGetValidationRequest request)
```
Comment (joelverhagen) [inline tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationProviderFacts.cs:168]: Shouldn't these methods be named `GetResponseAsync`?

File: src/Validation.Common.Job/Storage/ValidatorStatusExtensions.cs [hunk 4]
```diff
@@ -10,10 +10,10 @@ namespace NuGet.Jobs.Validation.Storage
     public static class ValidatorStatusExtensions
     {
         /// <summary>
-        /// Maps the provided validation status entity and its associated issues to a <see cref="IValidationResult"/>.
+        /// Maps the provided validation status entity and its associated issues to a <see cref="INuGetValidationResponse"/>.
         /// This method does not attempt to deserialize the issue data.
         /// </summary>
-        public static IValidationResult ToValidationResult(this ValidatorStatus validatorStatus)
+        public static INuGetValidationResponse ToValidationResponse(this ValidatorStatus validatorStatus)
```
Comment (joelverhagen) [inline src/Validation.Common.Job/Storage/ValidatorStatusExtensions.cs:16]: Thoughts on `ToNuGetValidationResponse`? We'll eventually have a method of the same name returning a unified type, right?

**Review by joelverhagen — APPROVED** [review]

File: tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationProviderFacts.cs (hunk 3 above)
Comment (loic-sharma) [reply tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationProviderFacts.cs:168]: `GetResultAsync` is the name that we accepted in the spec ([here](https://github.com/NuGet/Engineering/blob/master/Server.Specs/OrchestratorUnifiedValidations.md#new-orchestrator-interfaces)), but renaming to `GetResponseAsync` makes sense. Will do

File: src/Validation.Common.Job/Storage/ValidatorStatusExtensions.cs (hunk 4 above)
Comment (loic-sharma) [reply src/Validation.Common.Job/Storage/ValidatorStatusExtensions.cs]: Yup good idea, fixed

File: tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationProviderFacts.cs (hunk 3 above)
Comment (loic-sharma) [reply tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationProviderFacts.cs]: Talked offline and chose `GetResponseAsync`.

**Pull request merged**